lib/webrtc/rtcpeerconnection/chrome.js

```diff
@@ -553,10 +193,373 @@ function shimDataChannel(dataChannel) {
+      : {}, configuration);
+
+    util.interceptEvent(this, 'datachannel');
+    util.interceptEvent(this, 'signalingstatechange');
+
+    const sdpFormat = sdpUtils.getSdpFormat(newConfiguration.sdpSemantics);
+    let peerConnection = new RTCPeerConnection(newConfiguration, constraints);
+
+    Object.defineProperties(this, {
+      _appliedTracksToSSRCs: {
+        value: new Map(),
+        writable: true
+      },
+      _localStream: {
+        value: new MediaStream()
+      },
+      _peerConnection: {
+        value: peerConnection
+      },
+      _pendingLocalOffer: {
+        value: null,
+        writable: true
+      },
+      _pendingRemoteOffer: {
+        value: null,
+        writable: true
+      },
+      _rolledBackTracksToSSRCs: {
+        value: new Map(),
+        writable: true
+      },
+      _sdpFormat: {
+        value: sdpFormat
+      },
+      _senders: {
+        value: new Map()
+      },
+      _signalingStateLatch: {
+        value: new Latch()
+      },
+      _tracksToSSRCs: {
+        value: new Map(),
+        writable: true
+      },
+      localDescription: {
+        enumerable: true,
+        get: function() {
+          return this._pendingLocalOffer ? this._pendingLocalOffer : peerConnection.localDescription;
+        }
+      },
+      remoteDescription: {
+        enumerable: true,
+        get: function() {
+          return this._pendingRemoteOffer ? this._pendingRemoteOffer : peerConnection.remoteDescription;
+        }
+      },
+      signalingState: {
+        enumerable: true,
+        get: function() {
+          if (this._pendingLocalOffer) {
+            return 'have-local-offer';
+          } else if (this._pendingRemoteOffer) {
+            return 'have-remote-offer';
+          }
+          return peerConnection.signalingState;
+        }
+      }
+    });
+
+    let self = this;
+
+    peerConnection.addEventListener('datachannel', function ondatachannel(event) {
+      shimDataChannel(event.channel);
+      self.dispatchEvent(event);
+    });
+
+    peerConnection.addEventListener('signalingstatechange', function onsignalingstatechange() {
+      if (peerConnection.signalingState === 'stable') {
+        self._appliedTracksToSSRCs = new Map(self._tracksToSSRCs);
+      }
+      if (!self._pendingLocalOffer && !self._pendingRemoteOffer) {
+        self.dispatchEvent.apply(self, arguments);
+      }
+    });
+
+    peerConnection.ontrack = function ontrack() {
+      // NOTE(mroberts): adapter.js's "track" event shim only kicks off if we set
+      // the ontrack property of the RTCPeerConnection.
+    };
+
+    if (typeof RTCPeerConnection.prototype.addTrack !== 'function') {
+      peerConnection.addStream(this._localStream);
+    }
+    util.proxyProperties(RTCPeerConnection.prototype, this, peerConnection);
+
+    inherits(ChromeRTCPeerConnection, EventTarget);
+  }
+
+
+  // if (typeof RTCPeerConnection.addTrack !== 'function') {
```

Comment:
We can't have any if statements that are not wrapped in a constructor/function so I only used the first declaration of the removeTrack() function after following the status of this bug being resolved: https://bugs.chromium.org/p/chromium/issues/detail?id=860853 -- not sure if this means we no longer need the try catch block or if we still need multiple removeTrack()s